cmake/CLI11GeneratePkgConfig.cmake

```diff
@@ -4,4 +4,4 @@ else()
   configure_file("cmake/CLI11.pc.in" "CLI11.pc" @ONLY)
 endif()
 
-install(FILES "${PROJECT_BINARY_DIR}/CLI11.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+install(FILES "${PROJECT_BINARY_DIR}/CLI11.pc" DESTINATION "${CMAKE_INSTALL_DATADIR}/pkgconfig")
```

Comment:
This change is wrong at least when using cli11 in compiled mode. The binary artifact (`libCLI11`) is very much arch dependant.
Noticed while updating the vcpkg package.